run_scripts/2_fetch_raw_data.py

```diff
@@ -70,43 +69,78 @@
     credentials_bucket = storage_client.bucket(bucket_name)
     credentials_file = credentials_bucket.blob(blob_name)
-    rapid_pro_token = credentials_file.download_as_string().strip()
+    rapid_pro_token = credentials_file.download_as_string().strip().decode("utf-8")
     print("Downloaded Rapid Pro token.")
 
+    with open(uuid_table_path) as f:
+        phone_number_uuid_table = PhoneNumberUuidTable.load(f)
+
+    with open(TEST_CONTACTS_PATH) as f:
+        test_contacts = json.load(f)
+
+    IOUtils.ensure_dirs_exist(f"{root_data_dir}/Raw Data")
+
+    rapid_pro = RapidProClient(rapid_pro_domain, rapid_pro_token)
+
+    # Load the previous export of contacts if it exists, otherwise fetch all contacts from Rapid Pro.
+    raw_contacts_path = f"{root_data_dir}/Raw Data/contacts_raw.json"
+    contacts_log_path = f"{root_data_dir}/Raw Data/contacts_log.jsonl"
+    try:
+        print(f"Loading raw contacts from file '{raw_contacts_path}'...")
+        with open(raw_contacts_path) as f:
+            raw_contacts = [Contact.deserialize(contact_json) for contact_json in json.load(f)]
+        print(f"Loaded {len(raw_contacts)} contacts")
+    except FileNotFoundError:
+        print(f"File '{raw_contacts_path}' not found, will fetch all contacts from the Rapid Pro server")
+        with open(contacts_log_path, "a") as f:
+            raw_contacts = rapid_pro.get_raw_contacts(raw_export_log=f)
+
     # Download all the runs for each of the radio shows
-    for show in SHOWS:
-        output_file_path = f"{root_data_dir}/Raw Data/{show}.json"
-        print(f"Exporting show '{show}' to '{output_file_path}'...")
-
-        exit_code = subprocess.call([
-            "./docker-run.sh",
-            "--flow-name", show,
-            "--test-contacts-path", TEST_CONTACTS_PATH,
-            rapid_pro_domain,
-            rapid_pro_token,
-            user,
-            "all",
-            uuid_table_path,
-            output_file_path
-        ], cwd=f"{rapid_pro_tools_dir}/fetch_runs")
-
-        assert exit_code == 0, f"{rapid_pro_tools_dir}/fetch_runs/fetch_runs.py failed with exit_code {exit_code}"
-
-    # Download all the runs for each of the surveys
-    for survey in SURVEYS:
-        output_file_path = f"{root_data_dir}/Raw Data/{survey}.json"
-        print(f"Exporting survey '{survey}' to '{output_file_path}'...")
-
-        exit_code = subprocess.call([
-            "./docker-run.sh",
-            "--flow-name", survey,
-            "--test-contacts-path", TEST_CONTACTS_PATH,
-            rapid_pro_domain,
-            rapid_pro_token,
-            user,
-            "latest-only",
-            uuid_table_path,
-            output_file_path
-        ], cwd=f"{rapid_pro_tools_dir}/fetch_runs")
-
-        assert exit_code == 0, f"{rapid_pro_tools_dir}/fetch_runs/fetch_runs.py failed with exit_code {exit_code}"
+    for flow in SHOWS + SURVEYS:
```

Comment:
We might consider SHOW_NAMES and SURVEY_NAMES --- they're not actually objects representing the shows themselves, or the data from them, but the name that we can use to get them

Reply:
also redundant due to moving these to configuration json